File: app/src/main/java/com/azure/minitwitter/retrofit/AuthTwitterService.java
package com.azure.minitwitter.retrofit;

import com.azure.minitwitter.retrofit.request.RequestCreateTweet;
import com.azure.minitwitter.retrofit.response.Tweet;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface AuthTwitterService {

    @GET("tweets/all")
    Call<List<Tweet>> getAllTweets();

    @POST("tweets/create")
    Call<Tweet> createTweet(@Body RequestCreateTweet requestCreateTweet);

}

File: app/src/main/java/com/azure/minitwitter/retrofit/AuthTwitterService.kt
package com.azure.minitwitter.retrofit

import com.azure.minitwitter.retrofit.request.RequestCreateTweet
import com.azure.minitwitter.retrofit.request.RequestUserProfile
import com.azure.minitwitter.retrofit.response.ResponseUploadPhoto
import com.azure.minitwitter.retrofit.response.ResponseUserProfile
import com.azure.minitwitter.retrofit.response.Tweet
import com.azure.minitwitter.retrofit.response.TweetDeleted
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.*

interface AuthTwitterService {

    // Tweets
    @GET("tweets/all")
    fun getAllTweets(): Call<List<Tweet>>

    @POST("tweets/create")
    fun createTweet(@Body requestCreateTweet: RequestCreateTweet): Call<Tweet>

    @POST("tweets/like/{idTweet}")
    fun likeTweet(@Path("idTweet") idTweet: Int): Call<Tweet>

    @DELETE("tweets/{idTweet}")
    fun deleteTweet(@Path("idTweet") idTweet: Int): Call<TweetDeleted>

    // Users
    @GET("users/profile")
    fun getProfile(): Call<ResponseUserProfile>

    @PUT("users/profile")
    fun updateProfile(@Body requestUserProfile: RequestUserProfile): Call<ResponseUserProfile>

    @Multipart
    @POST("users/uploadprofilephoto")
    fun uploadProfilePhoto(@Part("file\"; filename=\"photo.jpeg\" ") file: RequestBody): Call<ResponseUploadPhoto>
}